dd-sdk-android/src/main/kotlin/com/datadog/android/error/internal/DatadogExceptionHandler.kt

```diff
@@ -43,6 +52,19 @@ internal class DatadogExceptionHandler(
         )
         writer.write(log)
 
+        contextRef.get()?.let {
+            println("Trigger work manager task")
+            val constraints = Constraints.Builder()
+                .setRequiredNetworkType(NetworkType.CONNECTED)
+                .build()
+            val uploadWorkRequest = OneTimeWorkRequest.Builder(UploadWorker::class.java)
```

Comment:
What happens if the Worker starts and the Datadog SDK is not initialised ? We should make sure everything all the dependencies are optional in the Worker.

Also have you tested the scenario where you have 2 threads using the fileReader and sending batches in the same time ?

Reply:
Good question, I'll run some tests tomorrow